### PR TITLE
{bio}[GCC/9.3.0] breseq v0.35.2

### PR DESCRIPTION
**Pull request by mboisson**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 2 out of 2 (1 easyconfigs in this PR)
node373.hydra.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/a2f9c33401d3bc59fa8186be628fbd3d for a full test report.

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 2 out of 2 (1 easyconfigs in this PR)
node127.hydra.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz, Python 2.7.5
See https://gist.github.com/1f531c582509ba0592125ea00418fb6f for a full test report.

File: easybuild/easyconfigs/b/breseq/breseq-0.35.2-GCC-9.3.0.eb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+runtest = 'test'
+
+sanity_check_paths = {
+    'files': ['bin/breseq', 'bin/gdtools'],
+    'dirs': [],
+}
+
+# breseq --help exists with non-zero exit code, so use grep
+sanity_check_commands = ["breseq --help | grep 'Usage: breseq'"]
```
Comment (lexming) [inline easybuild/easyconfigs/b/breseq/breseq-0.35.2-GCC-9.3.0.eb:29]: It is also possible to use `breseq --version`, which returns a nice exit code 0
```suggestion
sanity_check_commands = ["breseq --version"]
```

**Review by lexming — CHANGES_REQUESTED** [review]
The build was successful but tests failed due to missing `R`
>!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!> FATAL ERROR <!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
== 2020-08-25 16:31:05,374 run.py:550 WARNING Found 3 errors in command output (output: ---> ERROR Could not determine version of installed executable "R".
    Failed check
    !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!> FATAL ERROR <!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!)

Apparently, any version of `R` will do: https://barricklab.org/twiki/pub/Lab/ToolsBacterialGenomeResequencing/documentation/installation.html#install-external-dependencies

**Comment by mboisson** [comment]
> The build was successful but tests failed due to missing `R`
> 
> > !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!> FATAL ERROR <!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
> > == 2020-08-25 16:31:05,374 run.py:550 WARNING Found 3 errors in command output (output: ---> ERROR Could not determine version of installed executable "R".
> > Failed check
> > !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!> FATAL ERROR <!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!)
> 
> Apparently, any version of `R` will do: https://barricklab.org/twiki/pub/Lab/ToolsBacterialGenomeResequencing/documentation/installation.html#install-external-dependencies

That's interesting, but EasyBuild does not have a way to specify "any version of R", and I'm against adding an artificial dependency over a given one.

File: easybuild/easyconfigs/b/breseq/breseq-0.35.2-GCC-9.3.0.eb (hunk 1 above)
Comment (mboisson) [reply easybuild/easyconfigs/b/breseq/breseq-0.35.2-GCC-9.3.0.eb:29]: That came from the existing recipe with `intel-2019a`. Should we strive for uniformity ?

**Comment by lexming** [comment]
@mboisson I do not think that `R` is an artificial dependency for `breseq`. It is clearly documented and tests fail if `R` is missing.

**Comment by mboisson** [comment]
R is not artificial, but a specific version is. It doesn’t make sense to force users to load a specific version of R when it would work with the one they need to use.
  

  
  

  
  
>

**Comment by mboisson** [comment]
The way we would solve this in Compute Canada would be to add a   
  
depends_on(“r”) (without version specifier)
  
In the modluafooter, but this is specific to a given MNS, and EasyBuild does not have a mechanism to do something similar
  

  
  

  
  
>

**Comment by lexming** [comment]
@mboisson I agree that such a feature would be useful in this case, but with current Easybuild we have to pick some version of `R` to fulfil this dependency. We cannot accept an easyconfig with missing dependencies.

**Pull request closed without merging**